kinder/ci/workflows/external-ca-tasks.yaml

```diff
@@ -0,0 +1,234 @@
+        docker cp $TMP_DIR/etcd/ca.crt $CLUSTER-control-plane-2:/etc/kubernetes/pki/etcd/ca.crt
+
+        # Copy shared kubeconfig files to the second CP Node
+        docker cp $TMP_DIR/admin.conf $CLUSTER-control-plane-2:/etc/kubernetes/admin.conf
+        docker cp $TMP_DIR/scheduler.conf $CLUSTER-control-plane-2:/etc/kubernetes/scheduler.conf
+        docker cp $TMP_DIR/controller-manager.conf $CLUSTER-control-plane-2:/etc/kubernetes/controller-manager.conf
+
+        # Copy the CA cert and key to the workers so that a kubeconfig.conf can be later generated
+        docker cp $TMP_DIR/ca.crt $CLUSTER-worker-1:/etc/kubernetes/pki/ca.crt
+        docker cp $TMP_DIR/ca.key $CLUSTER-worker-1:/etc/kubernetes/pki/ca.key
+        docker cp $TMP_DIR/ca.crt $CLUSTER-worker-2:/etc/kubernetes/pki/ca.crt
+        docker cp $TMP_DIR/ca.key $CLUSTER-worker-2:/etc/kubernetes/pki/ca.key
+
+        # Remove the temporary directory
+        rm -rf ${TMP_DIR}
+
+        # Generate remaining certificates and kubelet.conf for bootstrap.
+        # Using valid embedded credentials in a kubelet.conf makes the kubelet bootstrap this Node right away,
+        # without depending on the KCM to sign the rotatable Node client credentials. Normally users should either
+        # disable the kubelet client certificate rotation and/or rely on an external signer.
+        docker exec $CLUSTER-control-plane-2 kubeadm init phase certs all --control-plane-endpoint=$CPE --v=$VERBOSITY
+        docker exec $CLUSTER-control-plane-2 kubeadm init phase kubeconfig kubelet --control-plane-endpoint=$CPE --v=$VERBOSITY
+
+        # Generate a kubelet.conf on the workers for bootstrap
+        docker exec $CLUSTER-worker-1 kubeadm init phase kubeconfig kubelet --control-plane-endpoint=$CPE --v=$VERBOSITY
+        docker exec $CLUSTER-worker-2 kubeadm init phase kubeconfig kubelet --control-plane-endpoint=$CPE --v=$VERBOSITY
+
+        # Remove the ca.key from all Nodes.
+        # The KCM tolerates the --cluster-signing-key flag, pointing to a file that is missing.
+        docker exec $CLUSTER-control-plane-1 rm /etc/kubernetes/pki/ca.key
+        docker exec $CLUSTER-control-plane-2 rm /etc/kubernetes/pki/ca.key
+        docker exec $CLUSTER-worker-1 rm /etc/kubernetes/pki/ca.key
+        docker exec $CLUSTER-worker-2 rm /etc/kubernetes/pki/ca.key
+  - name: init
+    description: |
+      Initializes the Kubernetes cluster with version "kubernetesVersion"
+      by starting the bootstrap control-plane node
+    cmd: kinder
+    args:
+      - do
+      - kubeadm-init
+      - --name={{ .vars.clusterName }}
+      - --loglevel=debug
+      - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+    timeout: 5m
+  - name: join
+    description: |
+      joins a secondary control-plane and worker Nodes to the cluster
+    timeout: 5m
+    cmd: /bin/bash
+    args:
+      - -c
+      - |
+        #!/bin/bash
```

Comment:
[1] same as above; also, `kinder do kubeadm join` already supports discovery modes, so might be we can re-use some code

Reply:
It did not work and i need to investigate why.

Reply:
This could be signal of a regression

Reply:
the problem here is that even with file discovery, with kubelet cert rotation enabled the kubelet will write a kubelet.conf that points to files that do not exists as nothing signs the CSRs for them.
pre-populating the kubelet.conf file with embeded certs on the nodes, solves this.

the alternative here is to disable client cert rotation via KubeletConfiguration, which seems trickier.